discretize/CylMesh.py

```diff
@@ -26,36 +30,50 @@ class CylMesh(BaseTensorMesh, BaseRectangularMesh, InnerProducts, CylView):
     """
 
     _meshType = 'CYL'
-
     _unitDimensions = [1, 2*np.pi, 1]
 
     def __init__(self, h, x0=None, cartesianOrigin=None):
         BaseTensorMesh.__init__(self, h, x0)
-        assert self.hy.sum() == 2*np.pi, "The 2nd dimension must sum to 2*pi"
+        assert np.abs(self.hy.sum() - 2*np.pi) < 1e-10, (
+            "The 2nd dimension must sum to 2*pi"
+        )
         if self.dim == 2:
             print('Warning, a disk mesh has not been tested thoroughly.')
         cartesianOrigin = (np.zeros(self.dim) if cartesianOrigin is None
                            else cartesianOrigin)
-        assert len(cartesianOrigin) == self.dim, ("cartesianOrigin must be the "
-                                                  "same length as the dimension"
-                                                  " of the mesh.")
+        assert len(cartesianOrigin) == self.dim, (
+            "cartesianOrigin must be the same length as the dimension of the "
+            "mesh."
+        )
         self.cartesianOrigin = np.array(cartesianOrigin, dtype=float)
 
     @property
     def isSymmetric(self):
         return self.nCy == 1
 
     @property
+    def _ntNx(self):
+        if self.isSymmetric:
```

Comment:
eventually, I would like to remove this assumption (there is conceptually a node at the origin, even for the cyl symmetric case, we just remove it). For the cyl symmetric code though, this assumption is used pretty heavily at the moment, so I think it warrants a separate pr